src/test/java/rx/SingleTest.java

```diff
@@ -1353,4 +1386,364 @@ public String call() throws Exception {
         int numberOfErrors = retryCounter.getOnErrorEvents().size();
         assertEquals(retryCount, numberOfErrors);
     }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testTakeUntilSuccess() {
+        Subscription sSource = mock(Subscription.class);
+        Subscription sOther = mock(Subscription.class);
+        TestSingle source = new TestSingle(sSource);
+        TestSingle other = new TestSingle(sOther);
+
+        TestSubscriber<String> result = new TestSubscriber<String>();
+        Single<String> stringSingle = Single.create(source).takeUntil(Single.create(other));
+        stringSingle.subscribe(result);
+        other.sendOnSuccess("one");
+
+        result.assertError(CancellationException.class);
+        verify(sSource, times(1)).unsubscribe();
```

Comment:
No need for `times(1)` here and below, `verify(obj).call()` already verifies `times(1)`